### PR TITLE
Reject Shipment Flow

### PR DESCRIPTION
**Pull request by macrael**
## Description

This adds the reject shipment flow, extracting the cancel move panel and calling it from ShipmentInfo

## Reviewer Notes

ConfirmWithReasonButton is a mouthful, but I didn't come up with something better while writing this, happy for a better name. 

## Setup

testdatagen scenario 7 gets TSP data setup

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal Story](https://www.pivotaltracker.com/story/show/160112046) for this change
* [This Redux FAQ](https://redux.js.org/faq/organizingstate) plus talking to Jim convinced me that managing display state inside of ConfirmWithReasonButton was appropriate.

## Screenshots
screen shot 2018-09-11 at 11 57 45 am
screen shot 2018-09-11 at 11 57 29 am
screen shot 2018-09-11 at 11 57 42 am

### CODE REVIEW

File: pkg/models/shipment.go [hunk 1]
```diff
@@ -329,6 +329,31 @@ func FetchShipmentByTSP(tx *pop.Connection, tspID uuid.UUID, shipmentID uuid.UUI
...
+func saveShipmentAndOffer(db *pop.Connection, shipment *Shipment, offer *ShipmentOffer) (*Shipment, *ShipmentOffer, *validate.Errors, error) {
+	// wrapped in a transaction because if one fails this actions should roll back.
+	responseVErrors := validate.NewErrors()
+	var responseError error
+	db.Transaction(func(db *pop.Connection) error {
+		transactionError := errors.New("rollback")
+
+		if verrs, err := db.ValidateAndUpdate(shipment); verrs.HasAny() || err != nil {
+			responseVErrors.Append(verrs)
+			responseError = errors.Wrap(err, "Error changing shipment status to ACCEPTED")
```
Comment (chrisgilmerproj) [inline pkg/models/shipment.go]: We'll probably want to change `ACCEPTED` to a variable that is either `ACCEPTED` or `REJECTED`.

File: pkg/models/shipment.go [hunk 2]
```diff
@@ -329,6 +329,31 @@ func FetchShipmentByTSP(tx *pop.Connection, tspID uuid.UUID, shipmentID uuid.UUI
...
+			return transactionError
+		}
+		if verrs, err := db.ValidateAndUpdate(offer); verrs.HasAny() || err != nil {
+			responseVErrors.Append(verrs)
+			responseError = errors.Wrap(err, "Error changing shipment offer status to ACCEPTED")
```
Comment (chrisgilmerproj) [inline pkg/models/shipment.go]: Same comment here.

**Comment by macrael** [comment]
The create was definitely confusing to me, I think that for action apis the action name should stand in for the method. I think that the method is the action when there is no “action”. 

—MacRae

**Comment by macrael** [comment]
I've added an e2e test to make sure rejections work. It unfortunately is not re-runnable, so I'm thinking about alternative ways to setting up our test data rather than just the one shot generate command.

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 3]
```diff
@@ -412,4 +412,46 @@ func (e e2eBasicScenario) Run(db *pop.Connection, loader *uploader.Uploader) {
 	hhg4 := offer4.Shipment
 	hhg4.Move.Submit()
 	models.SaveMoveDependencies(db, &hhg4.Move)
+
+	/*
+	 * Service member with uploaded orders and an approved shipment to have weight added
+	 * This shipment is rejected by the e2e test.
+	 */
+	email = "hhg@addweigh.ts"
```
Comment (chrisgilmerproj) [inline pkg/testdatagen/scenario/e2ebasic.go]: Change the email here to be `hhg@reject.ed`.  Erin pointed out to me that if you need to log in as a user after a test ran to look at what happened its helpful to have unique emails.

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 4]
```diff
@@ -412,4 +412,46 @@ func (e e2eBasicScenario) Run(db *pop.Connection, loader *uploader.Uploader) {
...
+		ServiceMember: models.ServiceMember{
+			ID:            uuid.FromStringOrNil("f4e362e9-9fdd-490b-a2fa-1fa4035b8f0d"),
+			FirstName:     models.StringPointer("HHG"),
+			LastName:      models.StringPointer("Submitted"),
+			Edipi:         models.StringPointer("4444567890"),
+			PersonalEmail: models.StringPointer(email),
+		},
+		Move: models.Move{
+			ID:               uuid.FromStringOrNil("7fca3fd0-08a6-480a-8a9c-16a65a100db9"),
+			Locator:          "REJECT",
```
Comment (chrisgilmerproj) [inline pkg/testdatagen/scenario/e2ebasic.go]: 👍

**Pull request merged**